osu.Game/Graphics/DrawableDate.cs

```diff
@@ -13,13 +13,59 @@ namespace osu.Game.Graphics
     public class DrawableDate : OsuSpriteText, IHasTooltip
     {
         private readonly DateTimeOffset date;
+        private readonly String dateFormat;
+        private readonly String tooltipFormat;
 
         public DrawableDate(DateTimeOffset date)
         {
             AutoSizeAxes = Axes.Both;
             Font = "Exo2.0-RegularItalic";
 
             this.date = date.ToLocalTime();
+            // if date's format is not specified, set it to an empty string,
+            // so that later we will know to humanize it
+            dateFormat = "";
+            // if tooltip's format is not specified, set it to an empty string
+            // so later we will know to default to a default time format
+            tooltipFormat = "";
```

Comment:
Setting to empty string is unnecessary.

`string` is null by default, so you should use `Text = string.IsNullOrEmpty(dateFormat) ? A : B`